safe_transaction_service/history/services/reorg_service.py

```diff
@@ -77,37 +82,52 @@ def check_reorgs(self) -> Optional[int]:
         """
         :return: Number of the oldest block with reorg detected. `None` if not reorg found
         """
+        first_not_confirmed_block = (
+            EthereumBlock.objects.not_confirmed().order_by("number").first()
+        )
+        if not first_not_confirmed_block:
+            return None
         current_block_number = self.ethereum_client.current_block_number
-        to_block = current_block_number - self.eth_reorg_blocks
-        for database_block in (
-            EthereumBlock.objects.not_confirmed(to_block_number=to_block)
+        confirmation_block = current_block_number - self.eth_reorg_blocks
+        queryset = (
+            EthereumBlock.objects.since(first_not_confirmed_block.number)
             .only("number", "block_hash", "confirmed")
             .order_by("number")
-            .iterator()
-        ):
-            blockchain_block, blockchain_next_block = self.ethereum_client.get_blocks(
-                [database_block.number, database_block.number + 1],
-                full_transactions=False,
+        )
+        paginator = Paginator(queryset, per_page=self.eth_reorg_blocks_batch)
+        for page_number in paginator.page_range:
+            current_page = paginator.get_page(page_number)
+            database_blocks = []
+            block_numbers = []
+            for block in current_page.object_list:
+                database_blocks.append(block)
+                block_numbers.append(block.number)
+            blockchain_blocks = self.ethereum_client.get_blocks(
+                block_numbers, full_transactions=False
             )
-            if (
-                HexBytes(blockchain_block["hash"])
-                == HexBytes(blockchain_next_block["parentHash"])
-                == HexBytes(database_block.block_hash)
+
+            for database_block, blockchain_block in zip(
+                database_blocks, blockchain_blocks
             ):
-                logger.debug(
-                    "Block with number=%d and hash=%s is matching blockchain one, setting as confirmed",
-                    database_block.number,
-                    HexBytes(blockchain_block["hash"]).hex(),
-                )
-                database_block.set_confirmed()
-            else:
-                logger.warning(
-                    "Block with number=%d and hash=%s is not matching blockchain hash=%s, reorg found",
-                    database_block.number,
-                    HexBytes(database_block.block_hash).hex(),
-                    HexBytes(blockchain_block["hash"]).hex(),
-                )
-                return database_block.number
+                if HexBytes(blockchain_block["hash"]) == HexBytes(
+                    database_block.block_hash
+                ):
+                    logger.debug(
+                        "Block with number=%d and hash=%s is matching blockchain one, setting as confirmed",
```

Comment:
this log message is not true at all, only setting as confirmed if the next condition is true `if database_block.number <= confirmation_block` 
Maybe we can include the message inside the if block?

Reply:
You are right

Reply:
Done